libs/plugins/stream-client/src/lib/core/models/generate-state-from-resource.ts

```diff
@@ -25,13 +25,18 @@ export function generateStateFromResource(resource: any, schemas: ContributionSc
   const stages = (resource && resource.data && resource.data.stages) || [];
   const items = makeStageItems(stages, schemaDefs);
   const graph = makeGraphNodes(stages);
+  const metadata = resource.metadata || {
```

Comment:
Initializing the metadata with empty data like the "or" case, we might be adding the metadata with 
```
{
   input: [],
   output: []
}
```
when the metadata was supposed to be null / undefined. We might have to add logic to ignore the input / output or the whole metadata when they are empty while exporting the stream resource.